plugins/modules/net_tools/nmcli.py

```diff
@@ -1078,13 +1089,26 @@ def connection_update(self, nmcli_command):
 
         # Constructing the command.
         for key, value in options.items():
+            if key in self.WIFI_SEC_SECRET_OPTIONS:
+                self.edit_commands += ['set %s %s' % (key, value)]
+                continue
             if value is not None:
                 cmd.extend([key, value])
 
         return self.execute_command(cmd)
 
     def create_connection(self):
-        status = self.connection_update('create')
+        (rc, out, err) = self.connection_update('create')
+        if rc != 0:
+            raise NmcliModuleError(err)
+        else:
+            status = (rc, out, err)
+        if self.edit_commands:
+            (rc, out, err) = self.edit_connection()
+            if rc != 0:
+                raise NmcliModuleError(err)
+            else:
+                status = (rc, out, err)
```

Comment:
```suggestion
        status = self.connection_update('create')
        
        if status[0] == 0 and self.edit_commands:
            status = self.edit_connection()
```

Reply:
Alright, yes that makes sense. I added some additional tests as well to try and test the failure mode.